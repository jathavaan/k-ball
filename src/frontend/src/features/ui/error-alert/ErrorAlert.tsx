import { ErrorAlertProps } from "./errorAlert.types.ts";
import { StyledErrorAlert } from "./errorAlert.style.ts";

export const ErrorAlert = ({ message, ...props }: ErrorAlertProps) => {
  return (
    <StyledErrorAlert {...props} variant="filled" severity="error">
      {message}
    </StyledErrorAlert>
  );
};
