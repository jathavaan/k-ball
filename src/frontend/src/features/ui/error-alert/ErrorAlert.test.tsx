import { render, screen } from "@testing-library/react";
import { ErrorAlert } from "./ErrorAlert";

describe("Tests for ErrorAlert", () => {
  it("Should render the error message", () => {
    const mockProps = {
      message: "Test Error Message",
    };

    render(<ErrorAlert {...mockProps} />);

    expect(screen.getByText(mockProps.message)).toBeDefined();
  });

  it("Should match ErrorAlert snapshot", () => {
    const mockProps = {
      message: "Test Error Message",
    };

    const { asFragment } = render(<ErrorAlert {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
