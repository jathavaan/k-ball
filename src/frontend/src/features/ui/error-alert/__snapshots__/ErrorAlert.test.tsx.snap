// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorAlert > Should match ErrorAlert snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-filledError MuiAlert-filled css-psg8n3-MuiPaper-root-MuiAlert-root"
    role="alert"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1lzspe0-MuiSvgIcon-root"
        data-testid="ErrorOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      Test Error Message
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tests for ErrorAlert > Should match ErrorAlert snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-filledError MuiAlert-filled css-psg8n3-MuiPaper-root-MuiAlert-root"
    role="alert"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1lzspe0-MuiSvgIcon-root"
        data-testid="ErrorOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      Test Error Message
    </div>
  </div>
</DocumentFragment>
`;
